src/components/password-strength-field/password-strength.ts

```diff
@@ -0,0 +1,142 @@
+    });
+    this.retrieveTranslations();
+  }
+
+  /**
+     * Gets the translations to set programmatically
+     */
+    retrieveTranslations(): Promise<Map<string, string>> {
+      return this.bulkTranslateService.getTranslations([
+          "password-strength-component.fill-strong-password",
+          "password-strength-component.password-weak",
+          "password-strength-component.password-normal",
+          "password-strength-component.password-good",
+          "password-strength-component.password-excellent",
+      ]).then(data => {
+          this.translations = data;
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.fill-strong-password");
+          return data;
+      });
+  }
+
+  /**
+   * Toggles the boolean for the advanced content view
+   */
+  togglePasswordAdvanced() {
+    this.showPasswordAdvanced = !this.showPasswordAdvanced;
+  }
+  
+  /**
+   * Whenever there is a change
+   * @param changes Object with changes
+   */
+  ngOnChanges(changes: SimpleChanges) {
+    this.checkPassword();
+  }
+
+  /**
+   * Check the password
+   */
+  checkPassword() {
+    // When there is a password to check
+    if (this.password !== undefined && this.password !== null && this.password.length > 0) {
+      this.passwordStrength = this.passwordService.passwordStrength(this.password);
+      this.passwordStrengthScore = this.passwordStrength.score;
+      // Switch between the different scores to show an appropriate message. 
+      switch(this.passwordStrengthScore) {
+        case 0: 
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.fill-strong-password");
+          break;
+        case 1:
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.password-weak");
+          break;
+        case 2:
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.password-normal");
+          break;
+        case 3:
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.password-good");
+          break;
+        case 4:
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.password-excellent");
+          break;
+        default:
+          break;
+      }
+      // Reset the feedback list
+      this.passwordFeedbackWarning = [];
+      // Set the times for the different scenario's
+      this.onlineThrottlingMessage = this.passwordStrength.crack_times_display.online_throttling_100_per_hour;
+      this.onlineNoThrottlingMessage = this.passwordStrength.crack_times_display.online_no_throttling_10_per_second;
+      this.offlineSlowHashingMessage = this.passwordStrength.crack_times_display.offline_slow_hashing_1e4_per_second;
+      this.offlineFastHashingMessage = this.passwordStrength.crack_times_display.offline_fast_hashing_1e10_per_second;
+      let index = 0;
```

Comment:
Seeing how you need an index wouldn't it make more sense to rewrite the for loop to a simple index based for loop?